### ISSUE TITLE
Can't connect to remote via SSH keys

### ISSUE THREAD

**Issue by geoidesic** [Sequel-Ace/Sequel-Ace#1660]
- Sequel Ace Version (including build number): Version 4.0.3 Build 20042
- Sequel Ace Source (App Store/GitHub/Homebrew): Um... I don't know. Presumably App Store.
- macOS Version: 13.0.1 (22A400)
- Processor Type (Intel/Apple): Apple M1
- MySQL Version: Ver 8.0.31 for Linux on aarch64 (MySQL Community Server - GPL)
- macOS Localization: I don't know how to find this but I'm UK-based, so presumably en_GB

**Is Issue Present in [Latest Beta](https://github.com/Sequel-Ace/Sequel-Ace/releases)?**

**Description**
I recently re-installed my OS and dutifully exported my Sequel Ace config before doing so. I re-imported it afterwards and all hell broke loose. Firstly it lost all my passwords. So that sucked. Secondly it could no longer connect to my remote databases via SSH keys. (I can connect to the remote server via normal SSH using keys).

I tried various settings changes... pointing it at my .ssh folder e.g. on the Network tab of settings but that just seemed to make matters worse. I'm not sure what's going on as Sequel Ace always used to just work straight out the box.

**Steps To Reproduce**
1. Export config settings
2. Re-install Mac OS
3. Import config settings
4. Panic!

**Expected Behaviour**

It should connect to the database via SSH using my user account's ssh keys

**Related Issues**

#370

**Additional Context**

Here's the error message from Sequel Ace:

```
Used command:  /usr/bin/ssh -v -N -S none -o ControlMaster=no -o ExitOnForwardFailure=yes -o ConnectTimeout=10 -o NumberOfPasswordPrompts=3 -o UserKnownHostsFile="/Users/noeldacosta/.ssh/known_hosts" -F /Users/noeldacosta/.ssh/id_rsa -i /Users/noeldacosta/.ssh/id_rsa -o TCPKeepAlive=no -o ServerAliveInterval=60 -o ServerAliveCountMax=1 mnr-staging@88.208.212.34 -L 62762:127.0.0.1:3306 -L 62763:88.208.212.34:3306

OpenSSH_9.0p1, LibreSSL 3.3.6
debug1: Reading configuration data /Users/noeldacosta/.ssh/id_rsa
/Users/noeldacosta/.ssh/id_rsa: line 1: Bad configuration option: -----begin
/Users/noeldacosta/.ssh/id_rsa line 2: no argument after keyword "b3blbnnzac1rzxktdjeaaaaabg5vbmuaaaaebm9uzqaaaaaaaaabaaablwaaaadzc2gtcn"
/Users/noeldacosta/.ssh/id_rsa line 3: no argument after keyword "nhaaaaaweaaqaaayeawkexkfuxqj0uldswrg6bfmyj5ksyzcmcag+qtahze1vzyzmuklob"
/Users/noeldacosta/.ssh/id_rsa line 4: no argument after keyword "2nftdtt1cxh6v2gcwcrod4i67qyyfoxaxo+nh3t7jmepiuvoltwuh6uchqtagojl/of3nj"
/Users/noeldacosta/.ssh/id_rsa line 5: no argument after keyword "r0sdbj8axlkifg7pljy57njziqyq/m/1w/92cgbbt4rthk/r3e5sn8r+q7cce24dj2fydo"
/Users/noeldacosta/.ssh/id_rsa line 6: no argument after keyword "consir0rarngbgl9cgfg2x6vhhfksss32nuiijlycbfqdtuyczdzoj8g/cehasjosxagyg"
/Users/noeldacosta/.ssh/id_rsa line 7: no argument after keyword "vt/bavtmexsgky02layaukqoe1iwbgutdypdqs76jf+h0keayq81bogdrspwuwcywwurd3"
/Users/noeldacosta/.ssh/id_rsa line 8: no argument after keyword "2od6d9lyt8buysgl95wo3bvgkaju1wn4nxpmitmxe5ls7j0m/jtz3g6ebhbjsr6z+ks6ne"
/Users/noeldacosta/.ssh/id_rsa line 9: no argument after keyword "ifujcnqwpytloeudwzavfdchkfoknzikq66+e05deog4jmobry6tdghx/cbnkb2ijzy9jo"
/Users/noeldacosta/.ssh/id_rsa line 10: no argument after keyword "82iixdhdzxhlloln+ympdue9gcqjzajjxg2fjyabaaafoicvfuyalrvmaaaab3nzac1yc2"
/Users/noeldacosta/.ssh/id_rsa line 11: no argument after keyword "eaaagbamjbmzh1f6o9fjxbfqxum35so+srgm3dhabvqrqb8xnvwwmzfci6g9jrbxu7dqsy"
/Users/noeldacosta/.ssh/id_rsa line 12: no argument after keyword "eldhnfnk6heiuu0gmhtsqfzvjyd0+4zhqylfti08fb+rgh0lqikiy/zn9zsudlawy/ams5"
/Users/noeldacosta/.ssh/id_rsa line 13: no argument after keyword "cbruz5y2oezswyqmkvzp9cp/dniaw7ek0xyv69xoujfefqu3antua49hwhtgqduokdkwkz"
/Users/noeldacosta/.ssh/id_rsa line 14: no argument after keyword "rm4c/xibrtser4yrzeret9p7oois8narua07snm3wacfbv3bb2rcaelwbshr0/wqfu5nl7"
/Users/noeldacosta/.ssh/id_rsa line 15: no argument after keyword "bistnpqgafjkjhnylgrlew8jw0eu+oxfodjhmmkvnw6ia0uqvrlgmflrkxd9qa+g/zwe/g"
/Users/noeldacosta/.ssh/id_rsa line 16: no argument after keyword "1glbi/evjt21ycmo1nvp+j16tclzl3uzuu49dp47c94onm4qy0kes/plop3on7ixj6lqcr"
/Users/noeldacosta/.ssh/id_rsa line 17: no argument after keyword "s6hla1swfx3xb5h6jj2yikouvntoq3qioi5qaa8ukwxov/3g5ygdoo82psapniisxr3c1x"
/Users/noeldacosta/.ssh/id_rsa line 18: no argument after keyword "5s6c5/spqxbnvrgqo82iy8rthy2ggqaaaambaaeaaagbaien9vtvnaex1l9kcuhuchip80"
/Users/noeldacosta/.ssh/id_rsa line 19: no argument after keyword "k0osjagxv4tbwrp2auyxisou5ldqkxyfsgsvg5xu3ot6ughrltcwl3psapvql/s7nokocs"
/Users/noeldacosta/.ssh/id_rsa line 20: no argument after keyword "tzuxi6mbywzpup4iv5t5dsxsn9ua1xtiatzmdlfoy2qcu18v3xmg3fh6gemvupn3nvszla"
/Users/noeldacosta/.ssh/id_rsa line 21: no argument after keyword "7a6dycfxd1/9d3p1d3yxnkrhlbdyqe4wydcdofckef7qouvzjn7kbrxyqdvvgd03mqxuiv"
/Users/noeldacosta/.ssh/id_rsa line 22: no argument after keyword "v+2yv3e0singa7ekupcoeb8ji1judsbgrq/zj1pemkt4u+r/yjlc/ceogv9ahm0iqulnc1"
/Users/noeldacosta/.ssh/id_rsa line 23: no argument after keyword "gz00ndjcwjrhudeyihnmm6whfft+s0aiy3grysvulfsxupbyrwfvhzqjndkok+bhsjfter"
/Users/noeldacosta/.ssh/id_rsa line 24: no argument after keyword "xpnknlgpzqya9sqyy/r6nfuibqoe0b+ajwabqhmfusi6vvmqh4pe3uee9jy0j/h5rensvt"
/Users/noeldacosta/.ssh/id_rsa line 25: no argument after keyword "8sbg0djqgaxne5pmgp2vdwjpnydpomkswrcw4qcjge4gm2usgqtsei9v7z0bhzuwqukqaa"
/Users/noeldacosta/.ssh/id_rsa line 26: no argument after keyword "amagmdx0fjcbkytooio6zlmzawkygqkh7zg6dzrfxzxw8csbmwdxfetsjgyxoxqnknddze"
/Users/noeldacosta/.ssh/id_rsa line 27: no argument after keyword "tnnqiufu5znqomlql0cmi7vfrptid9m7nezirnd56orxc8mx/3ctvxcdftajj8wnbcbhxv"
/Users/noeldacosta/.ssh/id_rsa line 28: no argument after keyword "o9eiw1xyw6pbzqx+m8k9t8ormmn9lluyvkajtuhi7msfnrxlsp0qp9r1zofh/d8vh6di3o"
/Users/noeldacosta/.ssh/id_rsa line 29: no argument after keyword "uyid9x7gc0tnagotxgkiqlprfu0pckwttmkqqhzyzd1vwpqzgaaadbapirqibvff85rc3j"
/Users/noeldacosta/.ssh/id_rsa line 30: no argument after keyword "hazjdvmf0pinysxfneintausupdik5hkxkiluldh26fznwe3j5ddkb1cfoovljvqc4poj0"
/Users/noeldacosta/.ssh/id_rsa line 31: no argument after keyword "nm5tyxuoduosxnliob9vcmzcmq/w0guveb7cgbv/ynrz6rgpr9n/ersuev8/hulzhdlebn"
/Users/noeldacosta/.ssh/id_rsa line 32: no argument after keyword "ipw04afofnbpob7z9c1tpxnfva8ary/skn6a2s0qe9dbrcc4idnj/cqjkdtunfh+jwrhmx"
/Users/noeldacosta/.ssh/id_rsa line 33: no argument after keyword "0pqkfqlt1rk0hu3wu5bi5clz4999qwqwaaameazw8zxg89sovrxgjt+pabczskc5kaytha"
/Users/noeldacosta/.ssh/id_rsa line 34: no argument after keyword "xquwxy2edtxjfppmeqo1kw/oh9rrafhfp2buyaxdyav8bnadm2bbrgilhfrqqh6npzop90"
/Users/noeldacosta/.ssh/id_rsa line 35: no argument after keyword "eiyvjdofxzmn/w266vybfhab9vnwj7pwcl3imdqs2cs4ch5gyh9t8mjkqxnffcujtswhxv"
/Users/noeldacosta/.ssh/id_rsa line 36: no argument after keyword "gjb6z3ifazhk6lehdyek5qvmea4vudn8g7wyghejvxikbvrj53e6q5kvfz8kldgcdhaul2"
/Users/noeldacosta/.ssh/id_rsa line 37: no argument after keyword "o/mofvo96hzv2daaaai25vzwxkywnvc3rhqe5vzwxzlu1hy0jvb2stqwlylmxvy2fsaqid"
/Users/noeldacosta/.ssh/id_rsa: line 38: Bad configuration option: baugbw
/Users/noeldacosta/.ssh/id_rsa: line 39: Bad configuration option: -----end
/Users/noeldacosta/.ssh/id_rsa: terminating, 39 bad configuration options
The SSH Tunnel has unexpectedly closed.
```

**Comment by geoidesic** [comment]
Great. Just noticed that the error message has exposed my private ssh key. Nice one.

**Comment by geoidesic** [comment]
So I tried changing the settings to point at the public key instead but that didn't work either.
<img width="632" alt="Screenshot 2022-12-27 at 16 44 05" src="https://user-images.githubusercontent.com/4637995/209682540-183b39ee-4d89-494d-97af-71d796c4cef1.png">

<img width="812" alt="Screenshot 2022-12-27 at 16 44 21" src="https://user-images.githubusercontent.com/4637995/209682532-09f63f72-0b3a-4161-ba70-f1864242a271.png">

**Comment by Kaspik** [comment]
Similar issues: https://github.com/Sequel-Ace/Sequel-Ace/issues/1505, https://github.com/Sequel-Ace/Sequel-Ace/issues/1462, https://github.com/Sequel-Ace/Sequel-Ace/issues/1264, https://github.com/Sequel-Ace/Sequel-Ace/issues/1108

Did you check all these? It's definitely just a setup issue.

**Comment by geoidesic** [comment]
I also tried pointing the config file setting at `/etc/ssh/ssh_config` but that didn't work either:

<img width="812" alt="Screenshot 2022-12-27 at 16 49 02" src="https://user-images.githubusercontent.com/4637995/209683124-ef51be0f-2880-4aed-b43d-bd6daaee82e3.png">

It just asks for the SSH password in that case.

**Comment by geoidesic** [comment]
> Similar issues: #1505, #1462, #1264, #1108
> 
> Did you check all these? It's definitely just a setup issue.

I'm certain you're right but the problem is I don't understand how to configure it.

**Comment by geoidesic** [comment]
I've also tried creating an empty `config` file in `~/.ssh` and pointing the settings at that... but that made no difference. It's really not clear from the documentation how this is supposed to work. `~/.ssh/config` is not something that gets created when you use `ssh-keygen` so I have no idea what's supposed to be in it because I've never needed it before. It's a bit strange to me that this is suddenly an issue in my life when I've been happily using Sequel Ace for years without any trouble. What's changed?

**Comment by geoidesic** [comment]
Sorry if anything above seems overly direct :) I'm just a bit flustered because I'm working a deadline and suddenly not able to do my work and can't seem to fix the issue on an app that is normally a complete pleasure to work with. So unexpected, it's traumatic ;)

**Comment by geoidesic** [comment]
OK, got it working. Trick was twofold:

1. I was missing ssh config file, which #1264 explains how to fix. (Just `touch ~/.ssh/config`)
2. The default network settings don't work. I had to follow #1108 – i.e. manually set both the `SSH Config:` and `Known Hosts:` values.
3. I had to add my ssh files to the allowed files on the Files tab.

<img width="812" alt="Screenshot 2022-12-27 at 18 25 12" src="https://user-images.githubusercontent.com/4637995/209694056-cc1b22e8-cbcf-44f9-b0b5-9263ed84fcf3.png">
<img width="812" alt="Screenshot 2022-12-27 at 18 25 09" src="https://user-images.githubusercontent.com/4637995/209694065-28308f75-71a6-46fb-9041-d67998113181.png">
